modules/core/04-channel/keeper/upgrade.go

```diff
@@ -0,0 +1,87 @@
+	"github.com/cosmos/ibc-go/v7/modules/core/04-channel/types"
+	host "github.com/cosmos/ibc-go/v7/modules/core/24-host"
+)
+
+// ChanUpgradeInit is called by a module to initiate a channel upgrade handshake with
+// a module on another chain.
+func (k Keeper) ChanUpgradeInit(
+	ctx sdk.Context,
+	portID string,
+	channelID string,
+	chanCap *capabilitytypes.Capability,
+	propsedUpgradeChannel types.Channel,
+	counterpartyTimeoutHeight clienttypes.Height,
+	counterpartyTimeoutTimestamp uint64,
+) (uint64, error) {
+	channel, found := k.GetChannel(ctx, portID, channelID)
+	if !found {
+		return 0, errorsmod.Wrapf(types.ErrChannelNotFound, "port ID (%s) channel ID (%s)", portID, channelID)
+	}
+
+	if channel.State != types.OPEN {
+		return 0, errorsmod.Wrapf(types.ErrInvalidChannelState, "expected %s, got %s", types.OPEN, channel.State)
+	}
+
+	if !k.scopedKeeper.AuthenticateCapability(ctx, chanCap, host.ChannelCapabilityPath(portID, channelID)) {
+		return 0, errorsmod.Wrapf(types.ErrChannelCapabilityNotFound, "caller does not own capability for channel, port ID (%s) channel ID (%s)", portID, channelID)
+	}
+
+	if propsedUpgradeChannel.State != types.INITUPGRADE || propsedUpgradeChannel.Counterparty.PortId != channel.Counterparty.PortId ||
+		propsedUpgradeChannel.Counterparty.ChannelId != channel.Counterparty.ChannelId {
+		return 0, errorsmod.Wrap(types.ErrInvalidChannel, "proposed channel upgrade is invalid")
+	}
+
+	if strings.TrimSpace(propsedUpgradeChannel.Version) == "" {
+		return 0, errorsmod.Wrap(types.ErrInvalidChannelVersion, "channel version must be not be empty")
+	}
+
+	if !channel.Ordering.SubsetOf(propsedUpgradeChannel.Ordering) {
+		return 0, errorsmod.Wrap(types.ErrInvalidChannelOrdering, "channel ordering must be a subset of the new ordering")
+	}
+
+	upgradeSequence, found := k.GetUpgradeSequence(ctx, portID, channelID)
+	if !found {
+		upgradeSequence = 1
+		k.SetUpgradeSequence(ctx, portID, channelID, upgradeSequence)
+	} else {
+		upgradeSequence++
+		k.SetUpgradeSequence(ctx, portID, channelID, upgradeSequence)
+	}
+
+	upgradeTimeout := types.UpgradeTimeout{
+		TimeoutHeight:    counterpartyTimeoutHeight,
+		TimeoutTimestamp: counterpartyTimeoutTimestamp,
+	}
+
+	k.SetUpgradeTimeout(ctx, portID, channelID, upgradeTimeout)
+	k.SetUpgradeRestoreChannel(ctx, portID, channelID, channel)
+
+	return upgradeSequence, nil
+}
+
+// WriteUpgradeInitChannel writes a channel which has successfully passed the UpgradeInit handshake step.
+// An event is emitted for the handshake step.
+func (k Keeper) WriteUpgradeInitChannel(
+	ctx sdk.Context,
+	portID,
+	channelID string,
+	channelUpgrade types.Channel,
+) {
+	defer telemetry.IncrCounter(1, "ibc", "channel", "upgrade-init")
+
+	k.SetChannel(ctx, portID, channelID, channelUpgrade)
+	k.Logger(ctx).Info("channel state updated", "port-id", portID, "channel-id", channelID, "previous-state", "OPEN", "new-state", "INITUPGRADE")
```

Comment:
```suggestion
	k.Logger(ctx).Info("channel state updated", "port-id", portID, "channel-id", channelID, "previous-state", types.OPEN.String(), "new-state", types.INITUPGRADE.String())
```

Reply:
I noticed this too but I left it as is for consistency 😛 

I've updated it as per your suggestion and opened https://github.com/cosmos/ibc-go/issues/3335